compose/service.py

```diff
@@ -685,15 +685,30 @@ def get_volumes_from_names(self):
     # TODO: this would benefit from github.com/docker/docker/pull/14699
     # to remove the need to inspect every container
     def _next_container_number(self, one_off=False):
-        containers = filter(None, [
-            Container.from_ps(self.client, container)
-            for container in self.client.containers(
-                all=True,
-                filters={'label': self.labels(one_off=one_off)})
-        ])
+        containers = self._fetch_and_inspect(
+            all=True,
+            filters={'label': self.labels(one_off=one_off)}
+        )
         numbers = [c.number for c in containers]
         return 1 if not numbers else max(numbers) + 1
 
+    def _fetch_and_inspect(self, **fetch_options):
+        # Account for containers that might have been removed since we fetched
+        # the list.
+        def soft_inspect(container):
+            if container is None:
```

Comment:
Is this necessary? I don't think there's a case where `self.client.containers()` will return null entries

Reply:
I had this because `Container.from_ps()` could return None: https://github.com/docker/compose/blob/771a1382bdd47b010874b1b13fa626e12a687f62/compose/container.py#L32

But now that we updated this to use `from_id`, this isn't needed anymore :-)